app/src/main/java/tttTDD/Java/Board.java

```diff
@@ -85,6 +83,54 @@ private String[] getAvailableMoves(String[] board) {
         }
 
         return Arrays.copyOf(availableMovesArray.toArray(), availableMovesArray.size(), String[].class);
+//        return Arrays.stream(board).filter((space) -> isSpaceAvailable(board.index)).toArray(String[]::new);
     }
 
+    private String[] mapBoardMovesToSet(int[] set) {
+        String[] setBeingChecked = new String[set.length];
+
+        for (int i = 0; i < set.length; i++) {
+            setBeingChecked[i] = board[set[i]];
+        }
+
+        return setBeingChecked;
+    }
+
+    private int[][] createWinningSet(int boardSize) {
+        int[][] winningSets = new int[(boardSize * 2) + 2][boardSize];
+        int location = 0;
+        int k = 0;
+        int d = 0;
+
+        for(int i = 0; i < boardSize; i++) {
+            for(int j = 0; j < boardSize; j++) {
+                winningSets[i][j] = location;
+                location++;
+            }
+        }
+
+        for(int i = boardSize; i < boardSize*2; i++) {
+            for(int j = 0; j < boardSize; j++ ) {
+                winningSets[i][j] = winningSets[j][k];
+            }
+            k++;
+        }
+
+        for(int i = boardSize * 2 ; i < boardSize * 2 + 1; i++) {
+            for(int j = 0; j < boardSize; j++) {
+                winningSets[i][j] = winningSets[j][d++];
+            }
+            d--;
+        }
+
+
+        for(int i = boardSize * 2 + 1; i < boardSize * 2 + 2; i++) {
+            for (int j = 0; j < boardSize; j++) {
+                winningSets[i][j] = winningSets[j][d--];
+
+            }
+        }
+
+        return winningSets;
+    }
```

Comment:
This method is very complex, enough so that I cannot understand what it is doing. Some things that would help tremendously:

- consider breaking up into several private methods with smaller purposes
- where breaking up a large function like this is not expedient, commenting the beginning of a new section of the subroutine can aid in comprehension
- use descriptive variable names. i and j at least have the benefit of being very conventional, but I have no idea what k and d are supposed to represent